src/Merchants.php

```diff
@@ -51,26 +51,48 @@ public static function get_merchant() {
 
 		if ( ! empty( $merchant_id ) ) {
 
-			try {
-				$merchant = API\Base::get_merchant( $merchant_id );
-				if ( $saved_merchant_id !== $merchant_id ) {
-					Pinterest_For_Woocommerce()::save_data( 'merchant_id', $merchant['data']->id );
-				}
-			} catch ( \Throwable $th ) {
-				$merchant = false;
-			}
+			// Get merchant if a merchant id was found.
+			$merchant = self::get_merchant_object( $merchant_id );
 		}
 
 		if ( ! $merchant || ( 'success' !== $merchant['status'] && 650 === $merchant['code'] ) ) {  // https://developers.pinterest.com/docs/redoc/#tag/API-Response-Codes Merchant not found 650.
 			// Try creating one.
-			$merchant = API\Base::maybe_create_merchant();
-			if ( 'success' === $merchant['status'] ) {
-				Pinterest_For_Woocommerce()::save_data( 'merchant_id', $merchant['data']->id );
+			$response = self::update_or_create_merchant();
+
+			if ( ! $response['merchant_id'] ) {
+				throw new \Exception( __( 'Wrong response when trying to create or update merchant.', 'pinterest-for-woocommerce' ), 400 );
 			}
+
+			$merchant = self::get_merchant_object( $response['merchant_id'] );
 		}
 
-		if ( ! $merchant || 'success' !== $merchant['status'] ) {
-			throw new \Exception( __( 'Response error when trying create a merchant or get the existing one.', 'pinterest-for-woocommerce' ), 400 );
+		if ( ! $merchant || 'success' !== $merchant['status'] || ! $merchant['data'] ) {
+			throw new \Exception( __( 'Response error when trying create a merchant or update the existing one.', 'pinterest-for-woocommerce' ), 400 );
+		}
+
+		// Update merchant id if it is different from the stored in DB.
+		if ( $saved_merchant_id !== $merchant['data']->id ) {
+			Pinterest_For_Woocommerce()::save_data( 'merchant_id', $merchant['data']->id );
+		}
+
+		return $merchant;
+	}
+
+
+	/**
+	 * Get merchant object using the merchant ID.
+	 *
+	 * @param string $merchant_id The merchant ID.
+	 *
+	 * @return mixed|boolean
```

Comment:
We should only have one return value from this method, and we can allow the thrown exception to bubble up to the calling code. That would allow the method body to simply be `return Base::get_merchant( $merchant_id );`.

Or even better, eliminate this method and just call `Base::get_merchant( $merchant_id )` directly with exception handling 🙂